<a href="https://colab.research.google.com/github/PYEITSON/AcademiaEstadistica/blob/main/Copia_de_Inferencia_AD2022_Plantilla.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

<center>

![](https://github.com/MayraDemoss1/Estadistica/blob/main/inferencia.jpg?raw=true)

<center/>


---
###**Actividad 3.1.** Inferencia Estadística


**1. ¿Cuál es la estatura de los estudiantes ITSON?**

Suponer que se tiene la siguiente población de estudiantes.

In [ ]:
estatura_poblacional <- c(round(rnorm(8000, 172.0, 35 ), 1),round(rnorm(8000, 160.0, 45 ), 1) )
#head(estatura_poblacional)

Estimar la media poblacional.

In [ ]:
mean(estatura_poblacional)

[1] 165.4747

**2. Seleccionar muestras de la población.**

In [ ]:
media_muestral <- mean(sample(estatura_poblacional, 20))
media_muestral

[1] 162.405

###**Actividad 3.2.** Técnicas de muestreo

| $$Media$$       |$$Proporción$$|
|:----------------|:------------|
| Población infinita| Población infinita      |
|$$n=\left (\frac{z _{a/2} \sigma }{e} \right )^{2}$$       | $$n= pq\left (\frac{z _{\alpha /2} }{e}\right )^{2}$$        |
| Población finita | Población finita       |
| $$n= \frac{\left ( Z_{\alpha/2}\sigma\right)^{2}N}{\left ( Z_{\alpha/2}\sigma\right)^{2} +e^{2}(N-1)}$$| $$n= \frac{\left ( z_{\alpha /2} \right )^{2}pqN}{e^{2}(N-1) + \left ( z_{\alpha /2} \right )^{2} pq}$$       |


**Ejercicio 1.** Generar muestras utilizando el muestreo aleatorio simple.

In [ ]:
seleccion_datos <- sample(c(1:16000), 30)
head(seleccion_datos) #posición de las personas seleccionadas
muestra <- estatura_poblacional[seleccion_datos]
muestra


[1] 13775  1236  6185 11968  9168 15754

[1] 124.7 183.9 221.0  65.3 292.8 209.1 119.8 235.4 138.0 201.5 223.0 214.7
[13] 176.0 196.8 209.9 231.8 138.8 181.6 213.9 175.5 146.0 185.6 148.8 135.9
[25] 182.5 233.8 178.9 111.5 164.3 145.2

**Ejercicio 2.** Cambiar el tamaño de muestra y analizar que sucede con la probabilidad cuando $n$ va creciendo. Ejemplificar el lanzamiento de un dado.

In [ ]:
dados <- sample(c(1:6),10 , replace = T)
#barplot(table(dados)) #El porcentaje debe de estar alrededor de 1/6
#round(table(dados)/length(dados), 2)





**Ejercicio 3.** Generar muestras utilizando el muestreo aleatorio sistemático.

El muestreo sistemático requiere un punto de partida, el cuál es un número aleatorio $0<x<K$.

$K$ se calcula de la siguiente manera: $N/n$

In [ ]:
#Generar la población.
sexo <- sample(c("Hombre", "Mujer"), 16000, replace = T, prob= c(0.47, 0.53))


In [ ]:
# Calcular el valor de K
n <- 385
N <- 16000
k <- floor(N/n)
k

[1] 41

In [ ]:
# Establecer el punto de partida
punto_inicio <- sample(c(1:k), 1)
punto_inicio

[1] 8

In [ ]:
# Generar datos de la muestra
datos_muestra <- seq(punto_inicio, by = k, length.out = n)
head(datos_muestra)
muestra_sexo <- sexo[datos_muestra]
head(muestra_sexo)

[1]   8  49  90 131 172 213

[1] "Mujer"  "Mujer"  "Mujer"  "Hombre" "Hombre" "Hombre"

**Ejercicio 4.** Generar muestras utilizando el muestreo aleatorio estratificado.

Para realizar el muestreo aleatorio estratificado se requiere conocer el tamaño de $N$ , tamaño de $n$ y la población por estrato.

In [ ]:
# Establecer los estratos
mujer <- sexo[sexo == "Mujer"]
hombre <- sexo[sexo == "Hombre"]


In [ ]:
# Determinar la cantidad de muestra por estrato.
n = 385
n1 <- (length(mujer)/length(sexo))*n
n2 <- (length(hombre)/length(sexo))*n
n1
n2

[1] 203.7131

[1] 181.2869

In [ ]:
# Seleccionar la muestra utilizando el muestreo aleatorio simple.
datos_mujeres <- sample(c(1:8367), 199) # Se generan los números aleatorios
muestra_mujeres <- mujer[datos_mujeres]
datos_hombres <- sample(c(1:7633), 180) # Segeneran los números aleatorios
muestra_hombres <- hombre[datos_hombres]

head(muestra_mujeres)
head(muestra_hombres)

[1] "Mujer" "Mujer" "Mujer" "Mujer" "Mujer" "Mujer"

[1] "Hombre" "Hombre" "Hombre" "Hombre" "Hombre" "Hombre"

**Ejercicio 3.**Calcular el tamaño de la muestra necesario para estimar la estatura promedio de los estudiantes de Itson, generando un muestra piloto de 30 obseraciones, un nivel de confianza del 95% y error del 5%

$$ n = \frac{(Z_{\alpha/2})^{2}\sigma^{2}}{e^2} $$


In [ ]:
#Generar datos para la prueba piloto.
muestra_piloto <- sample(estatura_poblacional, 30)

#Estimarla varianza 
varianza <- var(muestra_piloto)

# Calcular el error
error <- mean(muestra_piloto)*0.05

# Calcular el valor valor Za/2 para un NC= 0.95
NC <- 0.95
Z <- qnorm(NC + (1 - NC)/2, 0, 1)
n <- (Z^2*varianza)/error^2
n



**Realizar la selección del tamaño de muestra.**

In [ ]:
muestra_estatura <- sample(estatura_poblacional, 15)
mean(muestra_estatura)

###**Actividad 3.3.** Estimación del tamaño de muestra.

**1. Estimar el tamaño de muestra para los diferentes valores que puede tomar la propoción.**

$$ n = \frac{(Z_{\alpha/2})^{2}\hat{p}(1-\hat{p})}{e^{2}}$$

In [ ]:
p <- seq(.1, .90, 0.1)
q <- 1 - p
e <- 0.05
NC <- .95
Z <- 1.96

n <- round((p*(1-p)*Z^2)/e^2, 1)
n

datos_p <- data.frame(p, q, n)


[1] 138.3 245.9 322.7 368.8 384.2 368.8 322.7 245.9 138.3

**2. Realizar un gráfico con las proporciones.**

In [ ]:
barplot(n, names.arg=p, ylim=c(0,400))

**Nivel de Confianza (NC)**

<center>

![](https://raw.githubusercontent.com/MayraDemoss1/Estadistica/e64ea769b61a5d985f8ca0e9d8e9993587a675e7/NC.png) 

<center/>



 

$ NC + \alpha = 1$, despejando $\alpha$  tenemos 
 
 $\alpha = 1 - NC $, como el nivel de confianza es el área sombreada, nos quedan dos zonas para el valor de $\alpha$ y dividimos entre 2, por lo tanto la probabilidad de la izquierda es $\frac{\alpha}{2}$: 
 
$$\frac{\alpha}{2} + NC + \frac{\alpha}{2} = 1$$

Si queremos saber el valor de Z, utilizamos la función normal inversa de la siguiente manera:

In [ ]:
NC <- 0.95
alfa_2 <- (1 - NC)/2
Z_alfa_2 <- qnorm(alfa_2, 0, 1)
Z_alfa_2

[1] -1.959964

Se obtuvo el valor de $Z_{\alpha/2}$ del lado izquierdo de la distribución es por eso que se obtuvo un valor negativo. Si queremos obtener el valor del lado derecho debemos de sumar el $NC$, por lo tanto tenemos:

In [ ]:
NC <- 0.95
alfa_2 <- (1 - NC)/2
Z_alfa_2 <- qnorm(NC + alfa_2, 0, 1)
Z_alfa_2

[1] 1.959964

**3. Utilizar software para realizar una prueba piloto y estimar la proporción de "Hombres".****

In [ ]:
seleccion_datos <- sample(c(1:16000), 30)
#head(seleccion_datos) #posición de las personas seleccionadas
muestra_p <- sexo[seleccion_datos]

**a) Calcular el valor de la proporción de la muestra piloto.**

In [ ]:
seleccion_datos <- sample(c(1:16000), 30)
#head(seleccion_datos) #posición de las personas seleccionadas
muestra_p <- sexo[seleccion_datos]

**b) Estimar el tamaño de muestra considerando un NC=95% y un error del 5%.**

In [ ]:
p <- length(muestra_p[muestra_p == "Hombre"])/length(muestra_p)
e <- 0.05
NC <- 0.95
alfa_2 <- (1 - NC)/2
Z_alfa_2 <- qnorm(NC + alfa_2, 0, 1)

n = (Z_alfa_2^2*p*(1-p))/e^2
n

**c) Realizar la selección de la muestra y calcular la proporción**

In [ ]:
muestra_s <- sample(sexo, 357)

p <- length(muestra_s[muestra_s == "Hombre"])/length(muestra_s)
p

ERROR: ignored

**4. Etimar el tamaño de muestra para los siguientes ejercicios:**

**a) Un fabricante de bombillas sabe que la desviación típica de la duración de las bombillas es 100 horas. Calcula el tamaño de la muestra que se ha de someter a prueba para tener una confianza del 95% de que el error de la duración media que se calcula sea menor a 10 horas. (n= 385 bombillas)**

**b) Para estimar la proporción de habitantes de una ciudad que poseen ordenador personal se toma una muestra de tamaño $n$. Calcula el valor mínimo de $n$ para garantizar, con un nivel de confianza del 95%, que el error de estimación no supera el 2%. (Como se desconoce la proporción, se hará partir del caso más desfavorable, que será 0,5). (n= 2401 habitantes)**

**c) Se desea realizar una encuesta entre la población juvenil de una determinada localidad para determinar la proporción de jóvenes que estaría a favor de una nueva zona de ocio. El número de jóvenes de dicha población es N=2000. Determinar el tamaño de muestra necesario para estimar la proporción de estudiantes que están a favor con un error de estimación de 0.05 y un nivel de confianza del 95%. (n= 323 jóvenes)**

**d) Queremos ajustar una máquina de refrescos de modo que el promedio del líquido dispensado quede dentro de cierto rango. La cantidad de liquido vertido por la máquina sigue una distribución normal con desviación estándar 0.15 decilitros. Deseamos que el valor estimado que se vaya a obtener comparado con el verdadero no sea superior a 0.02 decilitros con una confianza del 95%.¿De qué tamaño debemos escoger la muestra? (n= 217 refrescos)**